frontEnd/src/views/AdminApplicant.vue

```diff
@@ -64,153 +74,185 @@
-          }else if(!servertasks[task].isSubmitted && !servertasks[task].isApproved){
-            servertasks[task].status = "Incomplete"
-            servertasks[task].buttonTitle = "Mark Complete"
-          }
-          this.tasks.push(servertasks[task])
+    getUserByID(this.applicantID).then((res) => {
+      this.applicant = res.data;
+      let servertasks = res.data.tasks;
+      for (const task in servertasks) {
+        if (servertasks[task].isSubmitted && servertasks[task].isApproved) {
+          servertasks[task].status = "Complete";
+          servertasks[task].buttonTitle = "Mark Incomplete";
+        } else if (
+          servertasks[task].isSubmitted &&
+          !servertasks[task].isApproved
+        ) {
+          servertasks[task].status = "Requires Approval";
+          servertasks[task].buttonTitle = "Mark Complete";
+        } else if (
+          !servertasks[task].isSubmitted &&
+          !servertasks[task].isApproved
+        ) {
+          servertasks[task].status = "Incomplete";
+          servertasks[task].buttonTitle = "Mark Complete";
         }
-      });
-  },
-  data(){
-    return {
-        applicantID: '',
-        adminID : '',
-        applicant: [],
-        tasks: [],
-        selectedIndex: '',
-        notif: "",
-        snackbar: false,
-        downloadIcons: {
-          noUpload: "mdi-download-off-outline",
-          upload: "mdi-cloud-download",
-          uploadComplete: "mdi-cloud-check",
-      },
-        Headers: [
-          {
-            text: 'Task Name',
-            align: 'start',
-            sortable: false,
-            value: 'name',
-          },
-          {
-            text: 'Completion Status',
-            align: 'start',
-            value: 'status',
-          },
-          {
-            text: 'Download',
-            align: 'start',
-            sortable: false,
-            value: 'upload',
-          },
-          {
-            text: 'Change Status',
-            sortable: false,
-            align: 'middle',
-            value: 'buttonTitle'
-          }
-        ],
-        noActions: [
-          "BIG Profile",
-          "You are no BIG Deal :(",
-          "You are a BIG Deal!"
-        ],
+        this.tasks.push(servertasks[task]);
       }
+    });
+  },
+  computed: {
+    userType: function () {
+      return this.applicant.isCommunityMentor ? "Community Mentor" : "Education Mentor";
+    },
   },
-
   methods: {
-    async updateUser(applicant){
-      await updateUser(applicant).then(response => {
-          this.applicant = response.data;
-          this.$emit('emitToApp', response.data);
+    async updateUser(applicant) {
+      await updateUser(applicant).then((response) => {
+        this.applicant = response.data;
+        this.$emit("emitToApp", response.data);
       });
     },
-    
-    changeStatus: function(status, index) {
+    async toggleUserType() {
+      this.switchLoading = true;
```

Comment:
should this be wrapped in an api call through the API services? just wondering if we are sticking to API services or getting rid of it. I would slightly prefer to keep it as I think its best practise and would reduce the amount of refactoring we would have to do on the front end but I am willing to follow whatever we decide on.